homeassistant/components/ezviz/__init__.py

```diff
@@ -1 +1,123 @@
-"""Support for Ezviz devices via Ezviz Cloud API."""
+"""Support for Ezviz camera."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyezviz.client import EzvizClient
+from requests import ConnectTimeout, HTTPError
+
+from homeassistant.const import CONF_PASSWORD, CONF_REGION, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    ATTR_SERIAL,
+    CONF_FFMPEG_ARGUMENTS,
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+from .coordinator import EzvizDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+PLATFORMS = [
+    "binary_sensor",
+    "camera",
+    "sensor",
+    "switch",
+]
+
+
+async def async_setup(hass, config):
+    """Set up the Ezviz integration."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Ezviz from a config entry."""
+
+    if entry.data.get(ATTR_SERIAL):
+        hass.data[DOMAIN][entry.unique_id] = entry
```

Comment:
We don't really need to save the camera entries in hass.data. The entries are stored in the config entries registry and we have access to them via `hass.config_entries.async_entries`.

We can access that in the camera platform and check for matching entries.